src/hooks/usePipelineRunsV2.ts

```diff
@@ -241,6 +240,26 @@ export const usePipelineRunsV2 = (
     isLoadingTektonResults,
   ]);
 
+  const prevClusterLengthRef = React.useRef<number | null>(null);
+
+  React.useEffect(() => {
+    const currentLength = processedClusterData?.length ?? 0;
+    const clusterBecameEmpty =
+      prevClusterLengthRef.current !== null &&
+      prevClusterLengthRef.current > 0 &&
+      currentLength === 0;
+
+    if (clusterBecameEmpty) {
+      if (kubearchiveEnabled) {
+        void kubearchiveResult.refetch();
+      } else {
+        void trRefetch();
+      }
+    }
+
+    prevClusterLengthRef.current = currentLength;
+  }, [processedClusterData?.length, kubearchiveEnabled, kubearchiveResult, trRefetch]);
```

Comment:
I dont think we need to refetch here because `etchRunsRef` keeps the resources that were deleted from cluster. We need to make sure that the resource is present in `etcdRunsRef`.

Reply:
That's true, I didn't pay attention to the `etcdRunsRef` logic. I figured it out the issue and fixed it. Basically, inside the `runs` memo, we had this logic at the beginning:

```tsx
    if (!etcdRuns.length) {
      return etcdRuns;
    }
```

but it should be:

```tsx
    if (!etcdRuns) {
      return etcdRuns;
    }
```